mitiq/utils.py

```diff
@@ -147,6 +149,69 @@ def _are_close_dict(dict_a: Dict[Any, Any], dict_b: Dict[Any, Any]) -> bool:
     return True
 
 
+class CircuitMismatchException(Exception):
+    pass
+
+
+def _generate_pc_circuit(
+        qubits: Iterable,
+        depth: int,
+        gate: EigenGate) -> Circuit:
+    """
+    Generates a circuit which should be the identity. Given a rotation
+    gate R(param), it applies R(2 * pi / depth) depth times, resulting
+    in R(2*pi)
+
+    Args:
+        qubits: a list of qubits
+        depth: the length of the circuit to create
+        gate: the base gate to apply several times
+    Returns:
+        circuit: a pc circuit that can be used for profiling
+    """
+    rotation_angle = 2*np.pi / depth
+    moments: List[ops.Moment] = []
+    for _ in range(depth):
+        operations = []
+        num_qubits = gate().num_qubits()
+        if num_qubits != len(qubits):
+            raise CircuitMismatchException(
+                "Number of qubits does not match domain size of gate.")
+        operations.append(gate(exponent=rotation_angle)(*qubits))
+        moments.append(ops.Moment(operations))
+    return Circuit(moments)
```

Comment:
Checking the number of qubits only needs to happen once (outside the loop). Further, I think this can be simplified to a one-liner like

```python
def _generate_pc_circuit(...):
    if len(qubits) != cirq.num_qubits(gate):
        <raise error>

    return cirq.Circuit(gate(exponent=2 * np.pi / depth).on(*qubits) for _ in range(depth))
```

This will keep the same moment structure as the current function has.